### PR TITLE
Add `background-origin` utilities

### PR DESCRIPTION
**Pull request by davidluhr**
This PR adds support for the [`background-origin`](https://developer.mozilla.org/en-US/docs/Web/CSS/background-origin) property.

An example use case is a button with a gradient background and transparent border. This typically results in unexpected border colors appearing, but can be fixed with `background-origin: border-box;` in CSS (or `bg-origin-border` in Tailwind CSS with this PR).

Here's the API:

| Class | CSS |
| --- | --- |
| `bg-origin-border` |  `background-origin: border-box` |
| `bg-origin-padding` |  `background-origin: padding-box` |
| `bg-origin-content` |  `background-origin: content-box` |

Only `responsive` variants are enabled by default, but can be configured under the `backgroundOrigin` key in your `tailwind.config.js` file.

### CODE REVIEW

File: __tests__/fixtures/tailwind-output-flagged.css [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/*! tailwindcss v2.1.0 | MIT License | https://tailwindcss.com */
+/*! tailwindcss v2.1.1 | MIT License | https://tailwindcss.com */
```
Comment (bagsa) [inline __tests__/fixtures/tailwind-output-flagged.css:1]: []()

File: __tests__/fixtures/tailwind-output-important.css [hunk 2]
```diff
@@ -7790,6 +7790,18 @@ video {
...
+}
+
+.bg-origin-padding {
+  background-origin: padding-box !important;
+}
+
+.bg-origin-content {
+  background-origin: content-box !important;
+}
+
```
Comment (bagsa) [inline __tests__/fixtures/tailwind-output-important.css:7804]: **
```suggestion
**[]()**
```
**

File: __tests__/fixtures/tailwind-output-flagged.css (hunk 1 above)
Comment (bagsa) [reply __tests__/fixtures/tailwind-output-flagged.css:1]: ****

**Review by bagsa — CHANGES_REQUESTED** [review]

**Comment by adamwathan** [comment]
Well done Mr. Luhr, well done.

**Pull request merged**